fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/builder/ElementBuilder.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.datagen.v1.builder;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import it.unimi.dsi.fastutil.objects.Object2ObjectArrayMap;
+import it.unimi.dsi.fastutil.objects.Object2ObjectMap;
+import org.jetbrains.annotations.Nullable;
+import org.joml.Vector3d;
+
+/**
+ * Instantiate this class in order to provide any specific <code>elements</code> to a given block/item model JSON.
+ */
+public class ElementBuilder {
+	private final Vector3d from;
+	private final Vector3d to;
+	@Nullable
+	private final RotationBuilder rotation;
+	private final boolean shade;
+	private final Object2ObjectMap<FaceBuilder.Face, FaceBuilder> faces = new Object2ObjectArrayMap<>();
```

Comment:
Why not `HashMap` or `EnumMap`? Also, you generally don't want the `*ArrayMap` as they're implemented using an array.

Reply:
Force of habit by now under the assumption that `Object2*` is generally faster when dealing with custom objects.